client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -506,55 +506,61 @@ def build_template(
             keep_placeholders (bool): Add flag to placeholder data for
                 hosts to decide if they want to remove
                 placeholder after it is used.
-            create_first_version (bool): create first version of a workfile
-            workfile_creation_enabled (bool): If True, it might create
-                                              first version but ignore
-                                              process if version is created
+            create_first_version (bool): Create first version of a workfile.
+                 When set to True, this option initiates the saving of the
+                 workfile for an initial version. It will skip saving if
+                 a version already exists.
+            workfile_creation_enabled (bool): Whether we are creating a new
+                workfile. If False, we assume we explicitly want to build the
```

Comment:
This is very confusing to me. Why would `workfile_creation_enabled == False` mean that we want to build the template in the current scene? I suggest you first explain the condition when the variable is True and then explain the alternative

Reply:
tbh, I find the variable name confusing. 
Furtherly, BigRoy's comment inside the if statement describes how we are using it. 

https://github.com/ynput/ayon-core/blob/0faaadab51b0e7b74c483a32e406f90b49f45636/client/ayon_core/pipeline/workfile/workfile_template_builder.py#L537-L538

Reply:
I understand the code but I'm talking about the meaning behind the variable name and what it's doing. If you had to guess as an user, disabling the "workfile creation" (workfile_creation_enabled == False) it would mean something like the workfile is not going to be created no?

Reply:
I always try to avoid boolean variable names that are negations, they are harder to read

Reply:
apparently I'm not the only one https://www.serendipidata.com/posts/naming-guidelines-for-boolean-variables